errors/error.go

```diff
@@ -96,13 +99,13 @@ const separator = ": "
 //
 // In order to avoid duplicated messages, we erase the fields present in the
 // underlying error if already set with same value in this error.
-func E(args ...interface{}) error {
+func E(args ...interface{}) *Error {
```

Comment:
curious why this change is needed.

Reply:
I was comparing with the result of List.Errors and wanted to build Error instances, I could also have used the type directly instead of errors.E(), but then the other question would be, why keep it as error ? When a type implements an interface it is rare to return the interface type itself on the constructor, you usually just return the concrete type and that will work out fine anyway when it is used as the interface. So I changed more because I don't see strong reasons for it to be error, it seems to enforce a constraint that is not necessary.